src/Homepage/index.js

```diff
@@ -62,25 +66,60 @@ class Homepage extends React.Component {
     selectedLocation: undefined,
     searchResultLatlng: undefined,
     searchResult: undefined,
+    showFormSubmissionNotification: false,
   };
 
   componentDidMount() {
     api.query().then(response => {
       const { results: data, location } = response;
       this.props.setIPlocation(location);
+      let selectedLocation;
+      let selectedStoreName;
+
+      const queryParams = qs.parse(this.props.location.search, { ignoreQueryPrefix: true });
+      if ('submittedStore' in queryParams) {
+        const storeData = JSON.parse(atob(queryParams.submittedStore));
+        selectedLocation = { lat: storeData.Latitude, lng: storeData.Longitude }
+        selectedStoreName = storeData[FORM_FIELDS.STORE_NAME];
+      }
+
       this.setState({
-        results: this.formatResults(data, location),
+        results: this.formatResults(data, selectedLocation || location),
```

Comment:
can you merge from master and update this? i updated the logic of where the `api.query` call is now made